src/NServiceBus.AcceptanceTests/Routing/AutomaticSubscriptions/When_handling_local_event.cs

```diff
@@ -0,0 +1,79 @@
+﻿namespace NServiceBus.AcceptanceTests.Routing.AutomaticSubscriptions
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_handling_local_event : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_autosubscribe_to_event()
+        {
+            var ctx = await Scenario.Define<Context>(x => x.Id = Guid.NewGuid())
+                .WithEndpoint<PublisherAndSubscriber>(b => b
+                .When((session, context) =>
+                {
+                    if (context.HasNativePubSubSupport)
+                    {
+                        context.EventSubscribed = true;
+                    }
+                    return Task.FromResult(0);
+                })
+                .When(c => c.EventSubscribed || c.HasNativePubSubSupport, (session, context) => session.Publish(new Event { ContextId = context.Id })))
+                .Done(c => c.GotEvent)
+                .Run().ConfigureAwait(false);
+
+            Assert.True(ctx.GotEvent);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public Guid Id { get; set; }
+            public bool GotEvent { get; set; }
+            public bool EventSubscribed { get; set; }
+        }
+
+        public class PublisherAndSubscriber : EndpointConfigurationBuilder
+        {
+            public PublisherAndSubscriber()
+            {
+                EndpointSetup<DefaultPublisher>(b =>
+                {
+                    // Make sure the subscription message isn't purged on startup
+                    b.PurgeOnStartup(true);
```

Comment:
so what do we do with this? Run it only in the core / with MSMQ and hope that other transports purge the queue at the same place if they do?

Should we maybe split out the assumptions about "purge on startup" into a dedicated test (which also includes messages from FST) and copy that test to all transports supporting PurgeOnStartup?